modules/phase_field/src/kernels/TWSDoubleObstacle.C

```diff
@@ -0,0 +1,35 @@
+#include "TWSDoubleObstacle.h"
+
+registerMooseObject("PhaseFieldApp", TWSDoubleObstacle);
+
+InputParameters
+TWSDoubleObstacle::validParams()
+{
+  InputParameters params = ADKernelValue::validParams();
+  params.addRequiredParam<std::string>("sigma", "the interface energy used in the kernel");
+  params.addRequiredParam<std::string>("eta", "the interface width used in the kernel");
+  params.addRequiredParam<std::string>("delta_g", "the driving force used in the kernel");
+  params.addClassDescription("traveling wave kenel for double obstacle potential,"
+                             "-gamma * (phi - 0.5) - sqrt(phi * (1 - phi)) * m.");
+  return params;
+}
+
+TWSDoubleObstacle::TWSDoubleObstacle(const InputParameters & parameters)
+  : ADKernelValue(parameters),
+    _sigma_name(getParam<std::string>("sigma")),
+    _sigma(getADMaterialProperty<Real>(_sigma_name)),
+    _eta_name(getParam<std::string>("eta")),
+    _eta(getADMaterialProperty<Real>(_eta_name)),
+    _delta_g_name(getParam<std::string>("delta_g")),
+    _delta_g(getADMaterialProperty<Real>(_delta_g_name))
```

Comment:
This is a bit of an anti-pattern. `getADMaterialProperty` actually takes the name of the input parameter as the argument (it just falls back to assuming it is the actual property name for backwards compatibility). You should either use `getADMaterialPropertyByName` if you pass in the property name, or better yet, don't get the property name if you don't need it!

```suggestion
    _sigma(getADMaterialProperty<Real>("sigma")),
    _eta(getADMaterialProperty<Real>("eta")),
    _delta_g(getADMaterialProperty<Real>("delta_g"))
```

(and again, those parameters _must_ use the correct derived string type. In this case `MaterialPropertyName`